src/Workspaces/CoreTest/CodeCleanup/ReduceTokenTests.cs

```diff
@@ -1824,6 +1824,137 @@ End Sub
+        Const s1 As UShort = 0060US
+        Const s2 As Short = &H0000FFFFS
+    End Sub
+End Module
+|]";
+
+            var expected = @"
+Module Program
+    Sub Main(args As String())
+        Const i0 As Integer = 60
+        Const i1 As Integer = 60%
+        Const i2 As Integer = &H6F
+        Const i3 As Integer = &O60
+        Const i4 As Integer = 60I
+        Const i5 As Integer = -60
+        Const i6 As Integer = 0
+        Const i7 As UInteger = 60UI
+        Const i8 As Integer = &HFFFFI
+        Const l0 As Long = 60L
+        Const l1 As Long = 60&
+        Const l2 As ULong = 60UL
+        Const s0 As Short = 60S
+        Const s1 As UShort = 60US
+        Const s2 As Short = &HFFFFS
+    End Sub
+End Module
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature,Traits.Features.ReduceTokens)]
+        public async Task ReduceIntegerLiteralWithNegativeHexOrOctalValue()
+        {
+            var code = @"[|
+Module Program
+    Sub Main(args As String())
+        Const s0 As Short = &HFFFFS
+        Const s1 As Short = &O177777S
+        Const s2 As Short = &H8000S
+        Const s3 As Short = &O100000S
+        Const i0 As Integer = &O37777777777I
+        Const i1 As Integer = &HFFFFFFFFI
+        Const i2 As Integer = &H80000000I
+        Const i3 As Integer = &O20000000000I
+        Const l0 As Long = &HFFFFFFFFFFFFFFFFL
+        Const l1 As Long = &O1777777777777777777777L
+        Const l2 As Long = &H8000000000000000L
+        Const l2 As Long = &O1000000000000000000000L
+    End Sub
+End Module
+|]";
+
+            var expected = @"
+Module Program
+    Sub Main(args As String())
+        Const s0 As Short = &HFFFFS
+        Const s1 As Short = &O177777S
+        Const s2 As Short = &H8000S
+        Const s3 As Short = &O100000S
+        Const i0 As Integer = &O37777777777I
+        Const i1 As Integer = &HFFFFFFFFI
+        Const i2 As Integer = &H80000000I
+        Const i3 As Integer = &O20000000000I
+        Const l0 As Long = &HFFFFFFFFFFFFFFFFL
+        Const l1 As Long = &O1777777777777777777777L
+        Const l2 As Long = &H8000000000000000L
+        Const l2 As Long = &O1000000000000000000000L
+    End Sub
+End Module
+";
+            await VerifyAsync(code, expected);
+
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.ReduceTokens)]
+        public async Task ReduceIntegerLiteralWithOverflow()
+        {
+            var code = @"[|
+Module Module1
+    Sub Main()
+        Dim sMax As Short = 0032768S
+        Dim usMax As UShort = 00655536US
+        Dim iMax As Integer = 002147483648I
+        Dim uiMax As UInteger = 004294967296UI
+        Dim lMax As Long = 009223372036854775808L
+        Dim ulMax As ULong = 0018446744073709551616UL
+        Dim z As Long = &O37777777777777777777777
+        Dim x As Long = &HFFFFFFFFFFFFFFFFF
+    End Sub
+End Module
+|]";
+
+            var expected = @"
```

Comment:
@brettfo the only difference in the strings is the first has the code markers in so that the test will work. I could re-write to:
```csharp
var code = @"<same contents as above>";
await VerifyAsync("[|" + code + "|]", code);
```